### PR TITLE
CoinMarketCap: New API

### PR DESCRIPTION
**Pull request by Boernsman**
nymea-plugins pull request checklist:

- [ x Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

**Comment by Boernsman** [comment]
Readme will be updated with the next documentation Pull-Request

File: coinmarketcap/integrationplugincoinmarketcap.cpp [hunk 1]
```diff
@@ -38,18 +38,62 @@ IntegrationPluginCoinMarketCap::IntegrationPluginCoinMarketCap()
 {
 }
 
+
+void IntegrationPluginCoinMarketCap::startPairing(ThingPairingInfo *info)
+{
+    NetworkAccessManager *network = hardwareManager()->networkManager();
+    QNetworkReply *reply = network->get(QNetworkRequest(QUrl("https://pro-api.coinmarketcap.com")));
+    connect(reply, &QNetworkReply::finished, this, [reply, info] {
```
Comment (mzanetti) [inline coinmarketcap/integrationplugincoinmarketcap.cpp:46]: s/this/info for 3rd arg

File: coinmarketcap/integrationplugincoinmarketcap.cpp [hunk 2]
```diff
@@ -38,18 +38,62 @@ IntegrationPluginCoinMarketCap::IntegrationPluginCoinMarketCap()
...
+        reply->deleteLater();
```
Comment (mzanetti) [inline coinmarketcap/integrationplugincoinmarketcap.cpp:47]: connect this individually

File: coinmarketcap/integrationplugincoinmarketcap.cpp [hunk 3]
```diff
@@ -38,18 +38,62 @@ IntegrationPluginCoinMarketCap::IntegrationPluginCoinMarketCap()
...
+{
+    Q_UNUSED(username)
+
+    QNetworkRequest request(QUrl("https://pro-api.coinmarketcap.com/v1/key/info"));
+    request.setRawHeader("X-CMC_PRO_API_KEY", secret.toUtf8());
+    request.setRawHeader("Accept", "application/json");
+
+    QNetworkReply *reply = hardwareManager()->networkManager()->get(request);
+    connect(reply, &QNetworkReply::finished, info, [this, reply, info, secret](){
+        reply->deleteLater();
```
Comment (mzanetti) [inline coinmarketcap/integrationplugincoinmarketcap.cpp:67]: connect this individually

**Pull request closed without merging**